### ISSUE TITLE
As a OSIO user I want to have a possibility to try workspace.next with Theia IDE 

### ISSUE THREAD

**Issue by ibuziuk** [openshiftio/openshift.io#3883]
`rhche` on osio is already build against upstream che `6.7.1` which means there is already a possibility to try [1]  workspace.next with Theia IDE  on openshift.io. It would be great to have a possibility to enable workspace.next with Theia IDE as a sidecar for a given set of osio users. We could probably start with toggle with `userID` strategy and once the whole workspace.next skeleton [2] would be mature enough switch to `enableByLevel` strategy (e.g. `alpha` / `beta` / `released`). This would allow to try workspace.next easier and got more feedback from users / PMs in terms of usability & feature gaps. Also, once it would be decided to switch everyone to workspace.next it would be just a matter of switching the toggle (workspace migration is not taken into account in this statement)

[1] https://drive.google.com/file/d/1x8jKFdHKilwD8r123i-quKueUMs5tbBO/view
[2] https://github.com/eclipse/che/issues/10123

**Comment by ibuziuk** [comment]
@garagatyi could you please add some input regarding this story ?

**Comment by garagatyi** [comment]
If we want to accomplish that we need:
1. Deploy marketplace placeholder that hosts Feature/ChePlugin yaml files.
1. In case toggle is enabled CheStarter would create a workspace and remove all the installers and somehow should find alternatives to the installers, so we need to find a solution for this issue.

What are the consequences for a user that has WS.NEXT toggle enabled:
1. WS.NEXT workspaces get created by CheStarter
1. Already existing workspaces would start and use old flow
1. New workspaces start in WS.NEXT flow
1. In case feature hosting is unavailable WS.NEXT workspaces don't start, but old ones do start
1. In case user switched back to old flow CheStarter would create traditional workspaces, but existing workspaces would behave as before (both traditional and WS.NEXT)

**Comment by ibuziuk** [comment]
issue description has been updated. 
@garagatyi would you be able to investigate `per workspace` toggle as part of - redhat-developer/rh-che#741 ?
@l0rd could we add this story to the walking skeleton epic  - eclipse/che#10123 ?

**Comment by gorkem** [comment]
Just to confirm, Does ws.next toggle means use ws.next way of creating/defining workspaces and use Theia IDE by default?

**Comment by ibuziuk** [comment]
> Does ws.next toggle means use ws.next way of creating/defining workspaces and use Theia IDE by default

@gorkem Correct. Also it is expected that this toggle would be `per workspace` (user should still have a possibility of creating classic / non-workspace.next workspaces via Codebases UI and UD in the general case

**Comment by ibuziuk** [comment]
After more detailed discussion with @garagatyi and @l0rd it appeared to be that workspace.next toggle enabled `per workspace` is not smth. that we are planning to implement. Basically, the plan is the following:

- there would be a feature toggle enabled for `all` osio users by default that would allow creation of workspace next (NOTE: when this toggle is enabled it just means that user `can` create a workspace next, but he also `can` create a classic workspaces). Why do we have this toggle at all if it is enabled for all by default ? Just in order to have a cleaner code in the upstream project and avoid `if` clause on the infrastructure level [1]

- next step would be making factories to support workspace next  https://github.com/eclipse/che/issues/10368 

- once factories would support `workspace.next`, the story `As a che.openshift.io user I want to have a possibility to try workspace.next with Theia IDE` would be resolved almost for free - it would be just a matter of creating a dedicated factory (workspace.next with Theia IDE running as a sidecar) and make it available.

[1] https://github.com/eclipse/che/pull/10092/files#diff-af3fecda517f34f356a133111f159465R112

**Comment by rhopp** [comment]
> there would be a feature toggle enabled for all osio users by default that would allow creation of workspace next

What about using `enableByLevel` and set the desired level to `experimental`? This way, only users with experimental features enabled will be able to start Theia workspaces.

**Comment by ibuziuk** [comment]
@rhopp currently we want to support workspace.next only via `che.openshift.io` flow NOT `openshift.io` flow (please, take a look at the issue description)
